protos/orchestrator_service.proto

```diff
@@ -289,6 +289,26 @@ message TerminateResponse {
     // No payload
 }
 
+message QueryInstancesRequest {
+    InstanceQuery query = 1;
+}
+
+message InstanceQuery{
+    repeated OrchestrationStatus runtimeStatus = 1;
+    google.protobuf.Timestamp createdTimeFrom = 2;
+    google.protobuf.Timestamp createdTimeTo = 3;
+    repeated google.protobuf.StringValue taskHubNames = 4;
+    int32 pageSize = 5;
```

Comment:
Let's rename this to `maxInstanceCount`. I feel like that's a more descriptive name.